src/clients/api/queries/useGetVrtAllowance.ts

```diff
@@ -0,0 +1,31 @@
+import { useQuery, QueryObserverOptions } from 'react-query';
+
+import getVrtAllowance, {
+  IGetVrtAllowanceInput,
+  GetVrtAllowanceOutput,
+} from 'clients/api/queries/getVrtAllowance';
+import FunctionKey from 'constants/functionKey';
+import { useTokenContract } from 'clients/contracts/hooks';
+
+type Options = QueryObserverOptions<
+  GetVrtAllowanceOutput,
+  Error,
+  GetVrtAllowanceOutput,
+  GetVrtAllowanceOutput,
+  FunctionKey.GET_VRT_TOKEN_ALLOWANCE
+>;
+
+const useGetVrtAllowanceIndex = (
+  { accountAddress, spenderAddress }: Omit<IGetVrtAllowanceInput, 'vrtTokenContract'>,
+  options?: Options,
+) => {
+  const vrtTokenContract = useTokenContract('vrt');
+
+  return useQuery(
+    FunctionKey.GET_VRT_TOKEN_ALLOWANCE,
+    () => getVrtAllowance({ vrtTokenContract, accountAddress, spenderAddress }),
+    options,
+  );
+};
+
+export default useGetVrtAllowanceIndex;
```

Comment:
Same thing here, I'd create a `useGetTokenAllowanceIndex` that's used for all tokens